### PR TITLE
Add TypeScript support

### LINKED ISSUE #68: Typescript?
I hope this isn't too much noise--I'm having some trouble stitching together Svelte Typescript support and I can not tell if it is a configuration issue or if I'm just ahead of support. Should this plugin be compatible with Typescript? 

I've tried the below eslintrc.js (and others). It appears eslint is still trying to parse the `<script lang="ts>` block with a Javascript parser.

```javascript
>>module.exports = {                                                               
    root: true,                                                                    
    parser: '@typescript-eslint/parser',                                           
    parserOptions: {                                                               
      ecmaVersion: 2018,                                                           
    },                                                                           
    env: {                                                                         
      node: true,                                                                  
    },                                                                             
    plugins: ['svelte3', '@typescript-eslint'],                                    
    extends: ['eslint:recommended'],                                               
    overrides: [                                                                   
      {                                                                            
        files: ['*.svelte'],                                                       
        processor: 'svelte3/svelte3',                                              
      },                                                                           
      {                                                                            
        files: ['**/*.ts', '**/*.svelte'],                                         
        env: {                                                                     
          node: false,                                                             
          es6: true,                                                               
        },                                                                         
        extends: [                                                                 
          'plugin:@typescript-eslint/eslint-recommended',                          
          'plugin:@typescript-eslint/recommended',                                 
        ],                                                                         
      },                                                                           
    ],                                                                             
  };     
```

If support is not ready, do you have an pointers on how I might try and contribute that?

### PR DESCRIPTION
**Pull request by dummdidumm**
Closes #68

This does not introduce a generic preprocessor-option. Instead it provides a TypeScript-specific implementation. The advantage is that we don't need hacks like deasync because transpilation and mapping can happen synchronously. It also makes post-processing and interaction with ESLint easier.

How it works:
1. transpile script contents from TS to JS
2. compile result to get Svelte compiler warnings
3. map those warnings back to its original positions
4. blank script contents
5. compile again to get the AST with original positions in the markdown part
6. use AST of step 5, vars of step 2 and blanked script content to produce the text which is handed to ESLint
7. Let the ESLint TypeScript plugin handle the TS code
8. adjust mappings

TODO:
- ~test/support for named blocks~
- ~check type-aware rules support~
- map suggestions (this could be a independent PR)

Question: Update `package-lock.json`? There are new (dev) dependencies and it would be good to have them in there. Why is it ignored in the first place?

### CODE REVIEW

**Review by nosovk — APPROVED** [review]

File: src/mapping.js [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+		return this.tag_info.original_start + offset;
+	}
+}
+
+class SourceMapper {
+	constructor(raw_source_map) {
+		this.raw_source_map = raw_source_map;
+	}
+
+	getOriginalPosition(generated_position) {
```
Comment (Conduitry) [inline src/mapping.js:46]: It's making me uneasy that the tests still passed even though the code in `preprocess.ts` has been changed to use `get_original_position` while this is still `getOriginalPosition`.

Comment (dummdidumm) [reply src/mapping.js:46]: This is a method called in line 126, so it's a different method (which I forgot to convert to _-style)

Comment (Conduitry) [reply src/mapping.js:46]: Ah, okay, thanks.

**Comment by dummdidumm** [comment]
Sorry about these code style issues, I'm no longer able to format myself since using Prettier 😄

**Comment by winfr34k** [comment]
Yeah!
Thanks to this getting merged, we can finally lint out TS/Svelte project 👍

**Pull request merged**